### ISSUE TITLE
[Install issue on Windows] Symbol’s value as variable is void: plantuml-java-args

### ISSUE THREAD

**Issue by badbob** [skuro/plantuml-mode#51]
``` Compiling file c:/Users/user/AppData/Roaming/.emacs.d/elpa/plantuml-mode-20161111.205/plantuml-mode.el at Tue Dec 20 15:58:43 2016
Entering directory ‘c:/Users/user/AppData/Roaming/.emacs.d/elpa/plantuml-mode-20161111.205/’
plantuml-mode.el:228:1:Error: Symbol’s value as variable is void: plantuml-java-args
```

But mode looks fully functional. No other issues was found.

**Comment by skuro** [comment]
I assume the line was triggered after `package-install` or something like that right? Just in case, to be able to reproduce the issue.

Also, when you say that the mode looks fully functional, did you also test `plantuml-preview`?

**Comment by badbob** [comment]
Yes, this line was triggered after _package-install_. 
Yes, _plantuml-preview_ works fine.

**Comment by Nebucatnetzer** [comment]
This bug happens on Debian Jessie as well.
However generating the UML works without a problem as far as I can tell..

**Comment by skuro** [comment]
Sorry for the latency, have been busy on multiple fronts and couldn't look into this issue any earlier. @badbob care to tell what version of Emacs are you running, native Windows version or cygwin?

**Comment by badbob** [comment]
My Emacs version:

```
GNU Emacs 25.1.1 (i686-w64-mingw32) of 2016-09-18
```
Warning reproduced as well: 

```
Compiling file c:/Users/user/AppData/Roaming/.emacs.d/elpa/plantuml-mode-20161225.1142/plantuml-mode.el at Thu May 11 12:21:09 2017
Entering directory ‘c:/Users/user/AppData/Roaming/.emacs.d/elpa/plantuml-mode-20161225.1142/’
plantuml-mode.el:229:1:Error: Symbol’s value as variable is void: plantuml-java-args
```

**Comment by jack2gs** [comment]
same with mine. Mac OS X 10.12.5 and Emacs  25.2.1

**Comment by c02y** [comment]
It happens to me too, `GNU Emacs 24.5.2 (i686-pc-linux-gnu, GTK+ Version 2.20.1) 
CentOS6.5 32-bit`

After `package-install` **plantuml-mode**, error message:

    plantuml-mode.el:229:1:Error: Symbol's value as variable is void: plantuml-java-args

But after I set `plantuml-jar-path` in my init.el and edit a .platuml file, `plantuml-mode` is working, I can use `C-c C-c` (plantuml-preview) to preview the file.

**Comment by leolle** [comment]
I am running Emacs 24.5.1, when I try package install plantuml-mode, I have the same issue.
`plantuml-mode.el:229:1:Error: Symbol's value as variable is void: plantuml-java-args`

**Comment by skuro** [comment]
It took a while but I finally tracked down the problem. The issue was due to `plantuml-java-args` being expanded in the `plantuml-start-process` macro, together with the fact that [`defcustom`s are not evaluated during byte compilation](https://emacs.stackexchange.com/questions/3648/macro-with-defcustom-leads-to-symbols-value-as-variable-is-void-error-when-by). The fix is also released as `v1.2.5`, soon coming in MELPA.
